packages/maskbook/src/extension/options-page/DashboardDialogs/Backup.tsx

```diff
@@ -36,9 +36,13 @@ const useDatabaseStyles = makeStyles((theme: Theme) =>
 export function DashboardBackupDialog(props: WrappedDialogProps) {
     const { t } = useI18N()
     const classes = useDatabaseStyles()
-    const { enqueueSnackbar, closeSnackbar } = useSnackbar()
+    const { enqueueSnackbar } = useSnackbar()
 
     const { value, loading } = useAsync(() => Services.Welcome.generateBackupJSON())
+
+    // since Android doesn't support `browser.download.downloads`,
+    //  we should create download url before click button.
+    const backupInfo = useAsync(() => Services.Welcome.createBackupUrl({ download: false, onlyBackupWhoAmI: false }))
```

Comment:
Isn't it costly?

Reply:
That's right, it cost. But what else can we do?

Reply:
this costy op only used in android but it will be run on any platform